exposestrategy/ingress.go

```diff
@@ -79,6 +80,14 @@ func NewIngressStrategy(client *client.Client, encoder runtime.Encoder, domain s
 }
 
 func (s *IngressStrategy) Add(svc *api.Service) error {
+	fullHostName, err := s.createIngress(s.client, svc)
+	if err == nil {
```

Comment:
I'll be honest - this part seemed weird to me.

`s.patchService` returns an error or nil. It looks like they just structured it this way to avoid another return inside the if statement.

Is this common in GoLang?